output/stream.go

```diff
@@ -33,26 +33,59 @@ var green = color.New(color.FgGreen).SprintFunc()
 var yellow = color.New(color.FgYellow).SprintFunc()
 var red = color.New(color.FgRed).SprintFunc()
 
+// Summary contains an account of the test run
+type Summary struct {
+	Failed bool
+
+	SuccessAmount int
+	FailAmount    int
+	SkippedAmount int
+}
+
 // Stream results to the console, error at end if any fail
-func Stream(tests <-chan execute.TestResponse) error {
+func Stream(tests <-chan execute.TestResponse) (summary Summary, err error) {
 	failed := false
 	for test := range tests {
 		for _, result := range test.Results {
 			var statStr string
 			switch result.Status {
 			case execute.Success:
 				statStr = green("✓")
+				summary.SuccessAmount++
 			case execute.Skipped:
 				statStr = yellow("S")
+				summary.SkippedAmount++
 			default:
 				statStr = red("F")
 				failed = true
+				summary.FailAmount++
 			}
 			fmt.Printf("%v - %v - %v\n", statStr, test.TestCase, result.Output)
 		}
 	}
 	if failed == true {
-		return errors.New("one or more tests failed")
+		summary.Failed = true
```

Comment:
Going to consolidate `summary.Failed` with the `err` since checking whether a test failed is needed in multiple places, and I don't want to pass around this pair everywhere.